modules/machine-health-checks-about.adoc

```diff
@@ -6,26 +6,41 @@
 [id="machine-health-checks-about_{context}"]
 = About MachineHealthChecks
 
-MachineHealthChecks automatically repairs unhealthy Machines in a particular
-MachinePool.
+A MachineHealthCheck is a resource which allows users to define conditions under
+which Machines within a Cluster should be considered unhealthy and should be
+automatically remediated.
 
-To monitor machine health, you create a resource to define the
-configuration for a controller. You set a condition to check for, such as
-staying in the `NotReady` status for 15 minutes or displaying a permanent condition
-in the node-problem-detector, and a label for the set of machines to monitor.
+When defining a MachineHealthCheck, users specify conditions to check for and a
+timeout for each of the conditions.
+If any of these conditions is met on the Machine's Node for the duration of the timeout,
+the Machine will be remediated.
+The action of remediating a Machine will trigger a new Machine to be created,
+to replace the failed one.
+
+For example, you may set a condition such as staying in the `NotReady` state
+for 15 minutes or displaying a permanent condition in the node-problem-detector.
 
 [NOTE]
 ====
-You cannot apply a MachineHealthCheck to a machine with the master role.
+Please note that MachineHealthChecks currently *only* support Machines that are
+owned by a MachineSet.
+Please review the xref:#limitations-when-deploying-machinehealthchecks at the bottom of this page for
+full details of MachineHealthCheck limitations.
 ====
 
-The controller that observes a MachineHealthCheck resource checks for the status
-that you defined. If a machine fails the health check, it is automatically deleted
-and a new one is created to take its place. When a machine is deleted, you
-see a `machine deleted` event. To limit disruptive impact of the machine
-deletion, the controller drains and deletes only one node at a time. If there
-are more unhealthy machines than the `maxUnhealthy` threshold allows for in the
-targeted pool of machines, remediation stops so that manual intervention can take
-place.
+When a Machine is deleted, you will see a `machine deleted` event.
+To limit disruptive impact of the Machine deletion, if there are more unhealthy
+Machines than the `maxUnhealthy` threshold allows for in the targeted pool of Machines,
+remediation stops so that manual intervention can take place.
+See xref:../modules/machine-health-checks-resource#short-circuiting-machinehealthcheck-remediation for more details.
```

Comment:
I don't know if this xref is formatted correctly or is going to do what I expect to do, how can I check this?

Reply:
For point 2, I'm not quite following, do I need to move the `xref` or do I need to move the thing it's pointing to? Could you link to an example of where this has been done already perhaps?

Reply:
@JoelSpeed you would need to move the xref from the module file (`modules/machine-health-checks-about.adoc`) to one of the assembly files that reference it.

At the top of the module file, there are references to the assembly files that have `include` references to the module, as follows:

```
// Module included in the following assemblies:
//
// * machine_management/deploying-machine-health-checks.adoc
// * post_installation_configuration/node-tasks.adoc
```

You can add the xref in one or more of the assemblies, after the `include` line that refers to the module. For example, if you wanted the xref to be included in the `machine_management/deploying-machine-health-checks.adoc`, which is [this page](https://docs.openshift.com/container-platform/4.6/machine_management/deploying-machine-health-checks.html#machine-health-checks-short-circuiting_deploying-machine-health-checks), you could add it after the following existing `include` line:

```
include::modules/machine-health-checks-about.adoc[leveloffset=+1]

.Additional resources

* See xref:../machine_management/deploying-machine-health-checks.adoc#machine-health-checks-short-circuiting_deploying-machine-health-checks[Short-circuiting MachineHealthCheck remediation] for more information about short circuiting.
```
The cross reference is properly formatted in the above example and I have tested that it links correctly to the target section. It also sits in an `.Additional resources` section and is a bulleted item.

If you want to test your xrefs, you can build the library locally using `asciibinder build` from the top level directory in your cloned repo. See https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/tools_and_setup.adoc#install-asciibinder-and-dependencies for more details.

Once your local build has been created, you can review the HTML rendering for the relevant assembly in your feature branch. For example:

$ firefox _preview/openshift-enterprise/update-mhc-docs/machine_management/deploying-machine-health-checks.html

If you make further changes, you can run `asciibinder build` again and refresh the HTML page.

I hope this helps. Please let me know if you have any further questions on this. Please feel free to ping me on Slack if you would like to go through the details.

Reply:
@JoelSpeed i think this is the change we need to make here, https://github.com/elmiko/openshift-docs/commit/7a7782510fd209b524b22f7aa17ed07f24200ec1